core/src/impl/Kokkos_AnalyzePolicy.hpp

```diff
@@ -120,6 +120,17 @@ struct SetWorkTag {
                                 typename PolicyBase::work_item_property>;
 };
 
+// Partial specialization to drop void arguments
```

Comment:
Why did you chose that over `!std::is_void<T> ? SetWorkTag<Base, T> : Base` in `AnalyzePolicy`?